src/wp-content/themes/twentytwentytwo/inc/patterns/comments.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Comments block (for WP >= 6.1), or legacy Post Comments block.
+ */
+
+if ( WP_Block_Type_Registry::get_instance()->is_registered( 'core/comments' ) ) {
+	return array(
+		'title'      => __( 'Comments block', 'twentytwentytwo' ),
+		'content'    => '<!-- wp:comments -->
+						<div class="wp-block-comments"><!-- wp:comments-title /-->
+
+						<!-- wp:comment-template -->
+						<!-- wp:columns -->
+						<div class="wp-block-columns"><!-- wp:column {"width":"40px"} -->
+						<div class="wp-block-column" style="flex-basis:40px"><!-- wp:avatar {"size":40,"style":{"border":{"radius":"20px"}}} /--></div>
+						<!-- /wp:column -->
+
+						<!-- wp:column -->
+						<div class="wp-block-column"><!-- wp:comment-author-name {"fontSize":"small"} /-->
+
+						<!-- wp:group {"layout":{"type":"flex"},"style":{"spacing":{"margin":{"top":"0px","bottom":"0px"}}}} -->
+						<div class="wp-block-group" style="margin-top:0px;margin-bottom:0px"><!-- wp:comment-date {"fontSize":"small"} /-->
+
+						<!-- wp:comment-edit-link {"fontSize":"small"} /--></div>
+						<!-- /wp:group -->
+
+						<!-- wp:comment-content /-->
+
+						<!-- wp:comment-reply-link {"fontSize":"small"} /--></div>
+						<!-- /wp:column --></div>
+						<!-- /wp:columns -->
+						<!-- /wp:comment-template -->
+
+						<!-- wp:comments-pagination -->
+						<!-- wp:comments-pagination-previous /-->
+
+						<!-- wp:comments-pagination-numbers /-->
+
+						<!-- wp:comments-pagination-next /-->
+						<!-- /wp:comments-pagination -->
+
+						<!-- wp:post-comments-form /--></div>
+						<!-- /wp:comments -->',
+	);
+}
+return array(
+	'title'      => __( 'Post Comments block', 'twentytwentytwo' ),
+	'content'    => '<!-- wp:heading --><h2>' . __( 'Comments', 'twentytwentytwo' ) . '</h2><!-- /wp:heading --><!-- wp:post-comments /-->',
```

Comment:
I think I may have found an issue. This heading will display even if comments are closed and this will be an unexpected side effect to users.

Is it possible to check if comments are enabled before outputting the heading?
-t depends on if that information is available at the time (init) when the pattern is registered as we would need to check it per post /page. -I don't think it is available.

Even if we found a way, this would not update if the user saves the template, and then we would be back to not having a heading.

Reply:
I'm not sure how I like this heading at all. It wasn't there on the original theme, and we are adding it now because the post comments block has an h3 instead? What are we trying to solve, the hierarchy of headings on the page?

Reply:
Yes, the incorrect heading order is an accessibility issue

Reply:
but it's an issue on the new block, not the old one, right? I think that should be fixed on the block, not on the theme, since it's affecting more than just TT2. What does TT3 do about this?

Reply:
Ok, but we've had the problem in the old block for a year and are only doing something about it now. I think the fix for that is to actually make use of the new block, and if you want to make sure that people running older versions get the fix too, then fix the block. I think this added heading is causing bigger problems than it really solves.

Reply:
Sorry, I hadn't read the whole conversation on the original issue, I'll think more about this problem, see how we can solve it. I'm not too happy that this is a TT2 only solution, though.

Reply:
Since the accessibility team decided that this single issue was not a blocker for adding the accessbility-ready tag,
We could remove the heading and only change the comments block with the patterns.
It would then be after 6.1 since it is then not considered a bug.

Reply:
I think the heading can be used to add context if there is a comments form but no comments, or if there are comments and the comment form / comment submission is closed. But not if comments are closed and there are no comments.

Reply:
I don't think the theme should include a comments.php file.
I also don't think the old block should be updated and that change backported to older versions... WP doesn't do that, right? Only for security updates?

Reply:
What if we replace the h3 to an h2 using the `render_block` hook?